pkg/core/registry_bridge/register.go

```diff
@@ -41,20 +59,78 @@ func (r *Registry) RegisterSelf() error {
 		return fmt.Errorf("issue getting sp data: %v", err)
 	}
 
-	// contract returns 0 if endpoint not registered
-	if spID.Uint64() == 0 {
-		logger.Infof("node %s : %s not registered on mainnet", nodeAddress.Hex(), nodeEndpoint)
-		if r.config.Environment == "dev" || r.config.Environment == "sandbox" {
-			if err := r.registerSelfOnEth(); err != nil {
-				return fmt.Errorf("error registering onto eth: %v", err)
-			}
-			// call again but registered this time
-			return r.RegisterSelf()
+	serviceType, err := contracts.ServiceType(r.config.NodeType)
+	if err != nil {
+		return fmt.Errorf("invalid node type: %v", err)
+	}
+
+	info, err := spf.GetServiceEndpointInfo(nil, serviceType, spID)
+	if err != nil {
+		return fmt.Errorf("could not get service endpoint info: %v", err)
+	}
+
+	if info.DelegateOwnerWallet != nodeAddress {
+		return fmt.Errorf("node %s is claiming to be %s but that endpoint is owned by %s", nodeAddress.Hex(), nodeEndpoint, info.DelegateOwnerWallet.Hex())
+	}
+
+	ethBlock := info.BlockNumber.String()
+
+	nodeRecord, err := r.queries.GetNodeByEndpoint(ctx, nodeEndpoint)
+	if err != nil {
+		r.logger.Infof("node %s not found on comet but found on eth, registering", nodeEndpoint)
+		if err := r.registerSelfOnComet(ethBlock, spID.String()); err != nil {
+			return fmt.Errorf("could not register on comet: %v", err)
 		}
-		logger.Info("continuing unregistered")
-		return nil
+		return r.RegisterSelf()
```

Comment:
doesn't this do the same loop as before?

Reply:
missed removing this